### PR TITLE
Williamhe/1394597 add oryx run script 2

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1394597
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl [hunk 1]
```diff
@@ -16,8 +16,18 @@ if [ "$SOURCE_DIR" == "$DESTINATION_DIR" ]
...
+
+	# we copy *.csproj to destination directory so the detector can identify
+	# the destination directory as a DotNet application
+	# when running oryx run-script
+	# 
+	# 2>/dev/null || :
+	# code snippet above is used to surpass cp error message & code
+	# since this is needed during: oryx run-script 
+	# but not during other dotnet builds
+	cp ${SOURCE_DIR}/*.csproj ${DESTINATION_DIR} 2>/dev/null || :
```
Comment (cormacpayne) [inline src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl:32]: nit: extra tab/spacing at the beginning of this section?

Comment (cormacpayne) [inline src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl:32]: nit: extra tab/spacing at the beginning of this section?

File: src/BuildScriptGenerator/VersionProviderHelper.cs [hunk 2]
```diff
@@ -55,5 +40,58 @@ internal static IEnumerable<string> GetVersionsFromDirectory(string versionsDir)
...
+
+            IEnumerable<DirectoryInfo> versionDirectories;
+            try
+            {
+                versionDirectories = Directory.EnumerateDirectories(versionsDir, "*", listOptions)
+                    .Select(versionDir => new DirectoryInfo(versionDir));
+            }
+            catch (IOException)
+            {
+                return (IEnumerable<DirectoryInfo>)Enumerable.Empty<string>();
```
Comment (cormacpayne) [inline src/BuildScriptGenerator/VersionProviderHelper.cs:91]: Is the casting necessary here? Any reason we can't just do the following?

```cs
return Enumerable.Empty<DirectoryInfo>();
```

File: src/startupscriptgenerator/src/hugo/config.go [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// --------------------------------------------------------------------------------------------
```
Comment (cormacpayne) [inline src/startupscriptgenerator/src/hugo/config.go:1]: Is there much different between the startup script generator we're adding for Go compared to Hugo, and between Go/Hugo and existing platforms?

**Review by cormacpayne — DISMISSED** [review]

File: src/startupscriptgenerator/src/hugo/main.go [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+	"flag"
+)
+
+func main() {
+	// setup flags
+	versionCommand := flag.NewFlagSet(consts.VersionCommandName, flag.ExitOnError)
+	
+	// setup env commands
+	setupEnvCommand := flag.NewFlagSet(consts.SetupEnvCommandName, flag.ExitOnError)
+	// setupEnvAppPathPtr := setupEnvCommand.String("appPath", ".", "The path to the application folder, e.g. '/home/site/wwwroot/'.")
```
Comment (qianz2) [inline src/startupscriptgenerator/src/hugo/main.go:20]: nit: remove comments

File: src/startupscriptgenerator/src/hugo/main.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+		viperConfig := common.GetViperConfiguration(fullAppPath)
+		configuration.EnableDynamicInstall = viperConfig.GetBool(consts.EnableDynamicInstallKey)
+		configuration.PreRunCommand = viperConfig.GetString(consts.PreRunCommandEnvVarName)
+		
+		entrypointGenerator := HugoStartupScriptGenerator{
+			AppPath:                  fullAppPath,
+			UserStartupCommand:       *userStartupCommandPtr,
+			BindPort:                 *bindPortPtr,
+			DefaultAppPath:           defaultAppFullPath,
+			// DefaultAppModule:         *defaultAppModulePtr,
```
Comment (qianz2) [inline src/startupscriptgenerator/src/hugo/main.go:58]: nit: remove comments

File: src/startupscriptgenerator/src/hugo/config.go (hunk 3 above)
Comment (william-msft) [reply src/startupscriptgenerator/src/hugo/config.go:1]: Currently there are no differences. But as we begin to develop more we will would need separate startups.

File: src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl (hunk 1 above)
Comment (william-msft) [reply src/BuildScriptGenerator/DotNetCore/DotNetCoreBashBuildSnippet.sh.tpl]: Updated. Thanks!

File: src/startupscriptgenerator/src/hugo/main.go (hunk 4 above)
Comment (william-msft) [reply src/startupscriptgenerator/src/hugo/main.go]: Updated. Thanks!

File: src/BuildScriptGenerator/VersionProviderHelper.cs (hunk 2 above)
Comment (william-msft) [reply src/BuildScriptGenerator/VersionProviderHelper.cs]: Update has been made.

**Review by cormacpayne — APPROVED** [review]

**Pull request merged**